### PR TITLE
chore(Sidebar): introduce `onCollapse` callback

### PR DESCRIPTION
**Pull request by besirovic**
[TRN-4574]

### Description

On Talent Portal we're working on removing an option for collapsing the sidebar from our settings page and we want to keep only built-in collapse option of Sidebar but we want to keep an ability to store the state or sidebar talent choose on our Backend so we can retain it on next visit and display sidebar at same state your left in on previous visit.

For that purpose, we're introducing `onCollapse` callback which will allow us to trigger our API each time collapse button in Sidebar is clicked.

### How to test

- `Collapsible` story is modified to trigger `onCollapse` callback, check the temploy

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

**Breaking change**

- [x] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[TRN-4574]: https://toptal-core.atlassian.net/browse/TRN-4574?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/pink-experts-cheat.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@toptal/picasso': major
```
Comment (TomasSlama) [inline .changeset/pink-experts-cheat.md:2]: I believe this should be only a minor update since we only add new functionality

**Review by TomasSlama — APPROVED** [review]
Looks good 👍

File: .changeset/pink-experts-cheat.md (hunk 1 above)
Comment (besirovic) [reply .changeset/pink-experts-cheat.md:2]: Sure, will update it 👌

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: packages/picasso/src/PageSidebar/PageSidebar.tsx [hunk 2]
```diff
@@ -92,6 +94,7 @@ export const PageSidebar = forwardRef<HTMLDivElement, Props>(function Sidebar(
 
   const handleCollapseButtonClick = useCallback(() => {
     setIsCollapsed(previousState => !previousState)
+    onCollapse?.()
```
Comment (mkrl) [inline packages/picasso/src/PageSidebar/PageSidebar.tsx:97]: Looking at your use case - are you sure you don't want to have the current (or next) collapsed state being passed to this callback? If you need to have it sent to the backend, maybe you could benefit from something like this
```suggestion
    onCollapse?.(isCollapsed)
```

**Review by mkrl — APPROVED** [review]
Looks good, but I left one question (just for your own sanity, perhaps you could need it in your implementation).

File: packages/picasso/src/PageSidebar/PageSidebar.tsx (hunk 2 above)
Comment (besirovic) [reply packages/picasso/src/PageSidebar/PageSidebar.tsx:98]: Not really, in our case the initial value will be returned by BE and we will send to API the opposite value of it once talent collapse/extend the sidebar.

For our use case it's enough just to have a callback to fire when sidebar is changing.

**Pull request merged**